sdb/internal/repl.py

```diff
@@ -73,12 +73,21 @@ def __init__(self,
         self.histfile = ""
         readline.set_completer(REPL.__make_completer(vocabulary))
 
-    def enable_history(self, history_file: str = '~/.sdb_history') -> None:
+    def enable_history(self, history_file: str) -> None:
         self.histfile = os.path.expanduser(history_file)
         try:
             readline.read_history_file(self.histfile)
         except FileNotFoundError:
             pass
+        except PermissionError:
+            self.histfile = ""
+            print(
+                f"Warning: You don't have permissions to read {history_file} and\n"
+                "         the command history of this session won't be saved.\n"
+                "         Either change this file's permissions, recreate it,\n"
+                "         or use an alternate path with the SDB_HISTORY_FILE\n"
+                "         environment variable.")
+            return
```

Comment:
does this mean we won't get tab completion (e.g. for command names) when this occurs? I see the call to `parse_and_bind` below, but not sure if that call is specific to the history file or not, so I wanted to ask..

Reply:
Nice catch! You are right. This is due to a different refactoring that I did a while ago. This call shouldn't be in that function as it has nothing to do with the history file. I'll move it up in the __init__ function when we initialize the vocabulary of words of which we tab-complete.

Reply:
I tested the above and added it to commit A. PR updated - resolving this.